src/pages/index.js

```diff
@@ -2,85 +2,223 @@ import "./index.css";
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
-
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
-}
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo(data);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
-}
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      deleteConfirm.renderDelete(true);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      deleteConfirm.renderDelete(false);
+      deleteConfirm.close();
+    });
+
+const handleLikeClick = (card) => {
+  api
+    .toggleCardLike(card._id, card._isLiked)
+    .then((cardData) => {
+      card.getLikeStatus(cardData.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  function handleChangePicture(avatar) {
+    changePictureModal.renderLoading(true, "saving...", "save");
+    api
+      .setUserAvatar(avatar)
+      .then((data) => {
+        userInfo.setUserAvatar(data.avatar);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        changePictureModal.renderLoading(false, "saving...", "save");
+        changePictureModal.close();
+      });
+  }
+  // handleChangePicture
+);
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
-}
+changePictureModal.setEventListeners();
 
-function handleCardAddSubmit(data) {
-  console.log(data);
-  renderCard(data, cardListEl);
-  cardAddForm.reset();
-}
+profilePicture.addEventListener("click", () => {
+  changePictureModal.open();
+  pictureFormValidator.resetValidation();
+});
 
-/* listeners */
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true, "saving...", "save");
+    api
+      .saveUserData({ name: userData.title, about: userData.description })
+      .then((res) => {
+        setUserInfo(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        editProfileModal.renderLoading(false, "saving...", "save");
+        editProfileModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well